src/zimscraperlib/video.py

```diff
@@ -0,0 +1,189 @@
+        arg_list += [
+            "-qmin",
+            str(self.quality_range[0]),
+            "-qmax",
+            str(self.quality_range[1]),
+            "-vf",
+            f"scale='{self.ffmpeg_video_scale}'",
+            "-codec:v",
+            video_codec,
+            "-codec:a",
+            audio_codec,
+            "-max_muxing_queue_size",
+            str(max_muxing_queue_size),
+            "-threads",
+            str(threads),
+            "-cpu-used",
+            str(ffmpeg_cpu_used),
+        ]
+
+        arg_list += extra_ffmpeg_params
+        return arg_list
+
+
+class VidUtil(object):
+    default_video_formats = {
+        "mp4": {
+            "vcodec": "h264",
+            "acodec": "aac",
+            "params": ["-movflags", "+faststart"],
+        },
+        "webm": {"vcodec": "libvpx", "acodec": "libvorbis", "params": [],},
+    }
+
+    def __init__(
+        self, video_compression_cfg, video_format, recompress,
+    ):
+        self.video_format = video_format
+        self.video_compression_cfg = video_compression_cfg
+        self.recompress = recompress
+
+    def update(self, **kwargs):
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    def recompress_video(self, src_path, dst_path, delete_src=True, **kwargs):
+        if (
+            "audio_codec" in kwargs
+            and "video_codec" in kwargs
+            and "extra_ffmpeg_params" in kwargs
+        ):
+            audio_codec = kwargs["audio_codec"]
+            video_codec = kwargs["video_codec"]
+            extra_ffmpeg_params = kwargs["extra_ffmpeg_params"]
+            for key in ["audio_codec", "video_codec", "extra_ffmpeg_params"]:
+                del kwargs[key]
+        elif self.video_format in self.default_video_formats:
+            audio_codec = self.default_video_formats[self.video_format]["acodec"]
+            video_codec = self.default_video_formats[self.video_format]["vcodec"]
+            extra_ffmpeg_params = self.default_video_formats[self.video_format][
+                "params"
+            ]
+        else:
+            raise TypeError(
+                "The video format with which VidUtil is initialized requires audio_codec, video_codec and extra_ffmpeg_params to be passed as arguments"
+            )
+
+        tmp_path = src_path.parent.joinpath(f"video.tmp.{self.video_format}")
+        args = (
+            ["ffmpeg", "-y", "-i", f"file:{src_path}"]
+            + self.video_compression_cfg.to_ffmpeg_args(
+                audio_codec=audio_codec,
+                video_codec=video_codec,
+                extra_ffmpeg_params=extra_ffmpeg_params,
+                **kwargs,
+            )
+            + [f"file:{tmp_path}"]
+        )
+        logger.info(
+            f"recompress {src_path} -> {dst_path} video format = {self.video_format}"
+        )
+        logger.debug(nicer_args_join(args))
+        subprocess.run(args, check=True)
```

Comment:
will crash on error. you may want to do that in two steps to clean-up temp file if you're not using a context manager